Logging/LogHandler/FileLogHandler.cs

```diff
@@ -56,14 +78,61 @@ public class FileLogHandler : AbstractAnvilBase, ILogHandler
         /// </summary>
         public LogLevel MinimumLevel { get; set; } = LogLevel.Debug;
 
+        /// <summary>
+        /// The maximum file size in bytes before the file is rotated. Default is 10MB.
+        /// </summary>
+        public long RotateFileSizeLimit { get; set; } = 10L * 1024 * 1024;
+
+        /// <summary>
+        /// The maximum number of rotated files to keep before deleting the oldest. Default is 5.
+        /// Ex. If the limit is 5, up to 5 rotated files in addition to the original log file, will be kept. After reaching this limit,
+        /// the next time the log file is rotated, the oldest file will be deleted. If the limit is 0, no rotated files are kept,
+        /// i.e. when the log file is rotated, it is simply erased and starts over.
+        /// </summary>
+        public int RotateFileCountLimit { get; set; } = 5;
+
         /// <summary>
         /// Creates a `FileLogHandler` that can write to the given file path.
         /// </summary>
         /// <param name="path">The text file to output messages to.</param>
-        /// <param name="append">If true, the file will have new logs appended to it, otherwise the file is replaced.</param>
-        public FileLogHandler(string path, bool append = true)
+        /// <param name="logType">The <see cref="LogType"/> to use.</param>
+        /// <param name="shouldRotate">
+        /// If true, the file will be rotated when the size limit is reached. When a file is rotated, it has an index appended to it,
+        /// i.e. "log.txt" is renamed "log.1.txt", and a new "log.txt" is opened. If "log.1.txt" already exists, it is first renamed
+        /// "log.2.txt", and so on, rotating through log files. Once the file count limit is reached, the oldest file is deleted.
+        /// </param>
+        public FileLogHandler(string path, LogType logType = LogType.Append, bool shouldRotate = true)
```

Comment:
I initially had it set up that way, but things got messy because I wanted different defaults for append vs. replace - that's changed, but it's still weird if you do something like pass a file count limit but no size limit. I guess that's fine, it will still work as instructed.

About the public setter, I was thinking about the syntax:
```
new FileLogHandler("log.txt")
{
    RotateFileCountLimit = 10;
}
```
...and figured changing those values after it's running was no stranger than changing `MinimumLevel` or `IncludeTimestamp`. Is there a benefit to the others being configurable post-instantiation, or could they all be readonly settings?

Reply:
I'll leave the open question, but I've made this change anyway.

Reply:
> I initially had it set up that way, but things got messy because I wanted different defaults for append vs. replace - that's changed, but it's still weird if you do something like pass a file count limit but no size limit. I guess that's fine, it will still work as instructed.

It would be an odd configuration but wouldn't break anything. I think that's ok.

> About the public setter, I was thinking about the syntax:
> ...

Yea, I appreciate the syntax for sure. In later versions of C# we'll be able to mark the setters with `init` and this will work great.

I suppose the difference with `IncludeTimestamp` and `MinimumLevel` is that they're not risky runtime changes. Your logs will just come out looking different after the change.

I think there's definitely a case for allowing `MinimumLevel` to be changed at runtime. The need to change `IncludeTimestamp` is much less likely.